commands/delete-database.go

```diff
@@ -15,81 +17,176 @@ func init() {
 	reset := cli.Command{
 		Name:      "delete",
 		Aliases:   []string{"delete-database"},
-		Usage:     "Delete an imported database",
+		Usage:     "Delete imported database(s)",
 		ArgsUsage: "<database>",
 		Flags: []cli.Flag{
 			forceFlag,
 			configFlag,
+			allFlag,
+			matchFlag,
+			regexFlag,
+			dryRunFlag,
 		},
-		Action: func(c *cli.Context) error {
-			res := resources.InitResources(c.String("config"))
-			db := c.Args().Get(0)
-			if db == "" {
-				return cli.NewExitError("Specify a database", -1)
-			}
-
-			return deleteDatabase(db, res, c.Bool("force"))
-
-		},
+		Action: deleteDatabase,
 	}
 
 	bootstrapCommands(reset)
 }
 
 //deleteDatabase deletes a target database
-func deleteDatabase(database string, res *resources.Resources, forceFlag bool) error {
+func deleteDatabase(c *cli.Context) error {
+	res := resources.InitResources(c.String("config"))
+
+	// Different command flags
+	match := c.String("match")
+	regex := c.String("regex")
+	bulk := c.Bool("all")
+	force := c.Bool("force")
+	dryRun := c.Bool("dry-run")
+	var names []string
+	var err error
+
+	if (match != "") && (!bulk) {
+		// Getting multiple dbs via string search
+		bulk = true
+
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
+
+		// Find dbs with matching names
+		for _, db := range dbs {
+			if strings.Contains(db, match) {
+				names = append(names, db)
+			}
+		}
 
-	// get all database names
-	names, err := res.DB.Session.DatabaseNames()
+	} else if (regex != "") && (!bulk) {
+		// Getting multiple dbs via regex query
+		bulk = true
 
-	// check if database exists
-	dbExists := util.StringInSlice(database, names)
+		// Get DB list
+		dbs := res.MetaDB.GetDatabases()
 
-	// check if metadatabase record for database exists
-	mDBExists := util.StringInSlice(database, res.MetaDB.GetDatabases())
+		// Compile regex, check if it's valid
+		regq, err := regexp.Compile(regex)
+		if err != nil {
+			return cli.NewExitError(err.Error, -1)
+		}
+
+		// Find dbs with regex query
+		for _, db := range dbs {
+			matched := regq.MatchString(db)
+			if matched {
+				names = append(names, db)
+			}
+		}
+
+	} else {
+		// get all database names
+		names = res.MetaDB.GetDatabases()
+	}
 
-	// if no force flag, verify action with the user
-	if !forceFlag {
-		fmt.Print("Are you sure you want to delete database ", database, " [y/N] ")
+	// check if we have databases
+	if err != nil || len(names) == 0 {
+		return cli.NewExitError("Failed to find any databases", -1)
+	}
 
-		read := bufio.NewReader(os.Stdin)
+	// For single database deletion
+	if !bulk {
 
-		response, _ := read.ReadString('\n')
+		database := c.Args().Get(0)
+		if database == "" {
+			return cli.NewExitError("Specify a database or specify --all or -a for all databases", -1)
```

Comment:
Let's remove mention of the -a flag here since there's others we're not mentioning.